pkg/skaffold/server/server.go

```diff
@@ -34,7 +35,12 @@ import (
 	"github.com/GoogleContainerTools/skaffold/proto"
 )
 
-var srv *server
+var (
+	srv *server
+
+	// waits for 1 second before shutting down the server
+	secondTimeout = 1 * time.Second
```

Comment:
I'd rename this to `forceShutdownTimeout` - the idea is that if `s.GracefulStop()` doesn't halt in this time, we'll forcefully shutdown the server, right?

Reply:
also, is 1 Second seems tight. I'd raise it to 5 seconds potentially. Also why do we need this? Is this the improvement over David's PR - as s.GracefulStop sometimes hangs? Do we know why it hangs? Can we tackle that root cause instead of forced shutdown with an arbitrary timeout that might or might not work?

Reply:
I read your description - can we add a comment here as well why we are forcing? Just explain the same thing you wrote in the PR that if connections are alive the server never shuts down.